smtpd/smtpd.go

```diff
@@ -218,6 +219,16 @@ func (s *session) serve() {
 		switch line.Verb() {
 		case "HELO", "EHLO":
 			s.handleHello(line.Verb(), line.Arg())
+		case "STARTTLS":
+			if s.srv.StartTLS == nil {
+				s.sendlinef("502 5.5.2 Error: command not recognized")
+				continue
+			}
+			s.sendlinef("220 Ready to start TLS")
+			if err := s.handleStartTLS(); err != nil {
+				s.errorf("failed to start tls: %s", err)
+				s.sendSMTPErrorOrLinef(err, "550 ??? failed")
```

Comment:
1. Should not it be `s.sendSMTPErrorOrLinef(err, "403 4.7.0 TLS handshake failed")`?
2. Should not it be inside `handleStartTLS` just to be similar to other usages?